migration/sql-files/035-wit-to-use-uuid.sql

```diff
@@ -87,21 +87,28 @@ ALTER TABLE work_item_types DROP CONSTRAINT work_item_link_types_check_name_c_lo
 -- Update work item link types
 ------------------------------
 
-ALTER TABLE work_item_link_types ADD COLUMN source_type_id uuid NOT NULL REFERENCES work_item_types(id) ON DELETE CASCADE;
-ALTER TABLE work_item_link_types ADD COLUMN target_type_id uuid NOT NULL REFERENCES work_item_types(id) ON DELETE CASCADE;
+ALTER TABLE work_item_link_types ADD COLUMN source_type_id uuid REFERENCES work_item_types(id) ON DELETE CASCADE;
+ALTER TABLE work_item_link_types ADD COLUMN target_type_id uuid REFERENCES work_item_types(id) ON DELETE CASCADE;
 
 UPDATE work_item_link_types SET source_type_id = (SELECT id FROM work_item_types WHERE name = source_type_name);
 UPDATE work_item_link_types SET target_type_id = (SELECT id FROM work_item_types WHERE name = target_type_name);
 
 ALTER TABLE work_item_link_types DROP COLUMN source_type_name;
 ALTER TABLE work_item_link_types DROP COLUMN target_type_name;
 
+-- Add NOT NULL criteria 
+ALTER TABLE work_item_link_types ALTER COLUMN source_type_id SET NOT NULL;
+ALTER TABLE work_item_link_types ALTER COLUMN target_type_id SET NOT NULL;
+
 --------------------
 -- Update work items
 --------------------
 
 -- NOTE: The foreign key is new!
 ALTER TABLE work_items RENAME type TO type_old;
-ALTER TABLE work_items ADD COLUMN type uuid NOT NULL REFERENCES work_item_types(id) ON DELETE CASCADE;
+ALTER TABLE work_items ADD COLUMN type uuid REFERENCES work_item_types(id) ON DELETE CASCADE;
 UPDATE work_items SET type = (SELECT id FROM work_item_types WHERE name = type_old);
-ALTER TABLE work_items DROP COLUMN type_old;
\ No newline at end of file
+ALTER TABLE work_items DROP COLUMN type_old;
```

Comment:
I think yes. But we're not in the position to restore historical data, yet. In the future this is probably bad or at least has to be considered more carefully.